### PR TITLE
Remove unused PETSC option

### PR DESCRIPTION
**Pull request by KiralyAgi**
This removes an old PETSC option

### CODE REVIEW

**Review by bangerth — CHANGES_REQUESTED** [review]
This looks ok, but you have a lot of merge commits in your branch. Do you know how to get rid of these (and, why they are there in the first place)?

**Comment by bangerth** [comment]
Related to #3338.

**Comment by KiralyAgi** [comment]
Hi Wolfgang,

I agree, it is strange that there are so many commits. As far as I see, they all originated from the times I updated ASPECT from the upstream. I usually do git pull upstream and then git push origin master. When I pull from upstream it always creates a commit for merging. But I don't know why is this visible on this branch. Is there a way to get the updates without having a commit for each time?

**Comment by KiralyAgi** [comment]
Okay, solved it! I followed the advice from here: https://stackoverflow.com/questions/33502205/how-to-delete-merge-commits

File: source/simulator/core.cc [hunk 1]
```diff
@@ -1047,17 +1044,9 @@ namespace aspect
...
-#ifdef ASPECT_USE_PETSC
-    SparsityTools::distribute_sparsity_pattern(sp,
-                                               dof_handler.locally_owned_dofs_per_processor(),
-                                               mpi_communicator, introspection.index_sets.system_relevant_set);
 
     sp.compress();
 
-    system_matrix.reinit (system_partitioning, system_partitioning, sp, mpi_communicator);
-#else
-    sp.compress();
```
Comment (tjhei) [inline source/simulator/core.cc:1059]: Keep this line, please.

**Comment by KiralyAgi** [comment]
I couldn't figure out how to change back 1 deleted line, so I did it all over again, and now it should be updated with the correct lines deleted/not-deleted

**Review by tjhei — APPROVED** [review]
Looks good!

**Pull request merged**